File: 2145-grid-game/2145-grid-game.go
package main

import "math"

func gridGame(grid [][]int) int64 {
    m := len(grid[0])
    res := int64(math.MaxInt64)
    
    topSum := int64(0)
    for i := 0; i < m; i++ {
        topSum += int64(grid[0][i])
    }
    
    bottomSum := int64(0)
    
    for i := 0; i < m; i++ {
        topSum -= int64(grid[0][i])
        res = min(res, max(topSum, bottomSum))
        bottomSum += int64(grid[1][i])
    }
    
    return res
}

func min(a, b int64) int64 {
    if a < b {
        return a
    }
    return b
}

func max(a, b int64) int64 {
    if a > b {
        return a
    }
    return b
}

File: 2145-grid-game/2145-grid-game.java
class Solution {
    public long gridGame(int[][] grid) {
        int n = grid.length;
        int m = grid[0].length; 
        long res = Long.MAX_VALUE;
        long top_sum = 0;
        for (int i = 0; i < grid[0].length; i++)
            top_sum += grid[0][i];
        long bottom_sum = 0;
        for (int i = 0; i < m; ++i) {
            top_sum -= grid[0][i];
            res = Math.min(res, Math.max(top_sum, bottom_sum));
            bottom_sum += grid[1][i];
        }
        return res;
    }
}